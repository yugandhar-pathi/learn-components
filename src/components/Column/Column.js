const Column = ({children}) => {
    let columnStyle = {
        display: "flex",
        flexDirection:"column"
    }
    return(
        <div style={columnStyle}>{children}</div>
    )
}
export default Column;