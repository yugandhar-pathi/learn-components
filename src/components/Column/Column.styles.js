import styled from 'styled-components'
export const ColumnStyle = styled.div`
 display: flex;
 flex-direction: column;
`