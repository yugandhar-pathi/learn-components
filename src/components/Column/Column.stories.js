import React from "react";
import {Column} from "./Column";

export default {
    component : Column, 
}

export const spaceBetween = () => <Column ><p>First Column</p><p>Second Column</p></Column>
export const spaceAround = () => <Column ><p>First Column</p><p>Second Column</p></Column>
export const center = () => <Column ><p>Hello World</p></Column>