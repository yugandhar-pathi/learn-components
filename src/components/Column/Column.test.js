import {render, screen} from '@testing-library/react'
import { Column } from './Column'

test('Column component', ()=>{
    render(<Column dataTestId = "column-id">
                <p>Column one</p>
                <p>Column two</p>
            </Column>
        )
    expect(screen.getByTestId("column-id").textContent).toBe("Column oneColumn two")
})

test("snapshot for column component", () => {
    render(<Column dataTestId = "column-id"><p>Column one</p><p>Column two</p></Column>)
    expect(screen.getByTestId("column-id")).toMatchSnapshot()
})