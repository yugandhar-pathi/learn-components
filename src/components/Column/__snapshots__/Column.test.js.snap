// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot for column component 1`] = `
<div
  class="sc-bczRLJ cSQhoY"
  data-testid="column-id"
>
  <p>
    Column one
  </p>
  <p>
    Column two
  </p>
</div>
`;
